File: go/tenniskata/TennisGame2.go
package tenniskata

type tennisGame2 struct {
	P1point int
	P2point int

	P1res       string
	P2res       string
	player1Name string
	player2Name string
}

func TennisGame2(player1Name string, player2Name string) TennisGame {
	game := &tennisGame2{
		player1Name: player1Name,
		player2Name: player2Name}

	return game
}

func (game *tennisGame2) GetScore() string {
	score := ""
	if game.P1point == game.P2point && game.P1point < 4 {
		if game.P1point == 0 {
			score = "Love"
		}
		if game.P1point == 1 {
			score = "Fifteen"
		}
		if game.P1point == 2 {
			score = "Thirty"
		}
		score += "-All"
	}
	if game.P1point == game.P2point && game.P1point >= 3 {
		score = "Deuce"
	}

	if game.P1point > 0 && game.P2point == 0 {
		if game.P1point == 1 {
			game.P1res = "Fifteen"
		}
		if game.P1point == 2 {
			game.P1res = "Thirty"
		}
		if game.P1point == 3 {
			game.P1res = "Forty"
		}

		game.P2res = "Love"
		score = game.P1res + "-" + game.P2res
	}
	if game.P2point > 0 && game.P1point == 0 {
		if game.P2point == 1 {
			game.P2res = "Fifteen"
		}
		if game.P2point == 2 {
			game.P2res = "Thirty"
		}
		if game.P2point == 3 {
			game.P2res = "Forty"
		}

		game.P1res = "Love"
		score = game.P1res + "-" + game.P2res
	}

	if game.P1point > game.P2point && game.P1point < 4 {
		if game.P1point == 2 {
			game.P1res = "Thirty"
		}
		if game.P1point == 3 {
			game.P1res = "Forty"
		}
		if game.P2point == 1 {
			game.P2res = "Fifteen"
		}
		if game.P2point == 2 {
			game.P2res = "Thirty"
		}
		score = game.P1res + "-" + game.P2res
	}
	if game.P2point > game.P1point && game.P2point < 4 {
		if game.P2point == 2 {
			game.P2res = "Thirty"
		}
		if game.P2point == 3 {
			game.P2res = "Forty"
		}
		if game.P1point == 1 {
			game.P1res = "Fifteen"
		}
		if game.P1point == 2 {
			game.P1res = "Thirty"
		}
		score = game.P1res + "-" + game.P2res
	}

	if game.P1point > game.P2point && game.P2point >= 3 {
		score = "Advantage player1"
	}

	if game.P2point > game.P1point && game.P1point >= 3 {
		score = "Advantage player2"
	}

	if game.P1point >= 4 && game.P2point >= 0 && (game.P1point-game.P2point) >= 2 {
		score = "Win for player1"
	}
	if game.P2point >= 4 && game.P1point >= 0 && (game.P2point-game.P1point) >= 2 {
		score = "Win for player2"
	}
	return score
}

func (game *tennisGame2) SetP1Score(number int) {

	for i := 0; i < number; i++ {
		game.P1Score()
	}

}

func (game *tennisGame2) SetP2Score(number int) {

	for i := 0; i < number; i++ {
		game.P2Score()
	}

}

func (game *tennisGame2) P1Score() {
	game.P1point++
}

func (game *tennisGame2) P2Score() {
	game.P2point++
}

func (game *tennisGame2) WonPoint(player string) {
	if player == "player1" {
		game.P1Score()
	} else {
		game.P2Score()
	}
}

File: vbnet/Tennis/TennisGame2.vb
Public Class TennisGame2
    Implements ITennisGame

    Private p1point As Integer
    Private p2point As Integer
    Private p1res As String = ""
    Private p2res As String = ""
    Private player1Name As String
    Private player2Name As String

    Public Sub New(ByVal player1Name As String, ByVal player2Name As String)
        Me.player1Name = player1Name
        p1point = 0
        Me.player2Name = player2Name
    End Sub

    Public Function GetScore() As String Implements ITennisGame.GetScore
        Dim score = ""

        If p1point = p2point AndAlso p1point < 3 Then
            If p1point = 0 Then score = "Love"
            If p1point = 1 Then score = "Fifteen"
            If p1point = 2 Then score = "Thirty"
            score += "-All"
        End If

        If p1point = p2point AndAlso p1point > 2 Then score = "Deuce"

        If p1point > 0 AndAlso p2point = 0 Then
            If p1point = 1 Then p1res = "Fifteen"
            If p1point = 2 Then p1res = "Thirty"
            If p1point = 3 Then p1res = "Forty"
            p2res = "Love"
            score = p1res & "-" & p2res
        End If

        If p2point > 0 AndAlso p1point = 0 Then
            If p2point = 1 Then p2res = "Fifteen"
            If p2point = 2 Then p2res = "Thirty"
            If p2point = 3 Then p2res = "Forty"
            p1res = "Love"
            score = p1res & "-" & p2res
        End If

        If p1point > p2point AndAlso p1point < 4 Then
            If p1point = 2 Then p1res = "Thirty"
            If p1point = 3 Then p1res = "Forty"
            If p2point = 1 Then p2res = "Fifteen"
            If p2point = 2 Then p2res = "Thirty"
            score = p1res & "-" & p2res
        End If

        If p2point > p1point AndAlso p2point < 4 Then
            If p2point = 2 Then p2res = "Thirty"
            If p2point = 3 Then p2res = "Forty"
            If p1point = 1 Then p1res = "Fifteen"
            If p1point = 2 Then p1res = "Thirty"
            score = p1res & "-" & p2res
        End If

        If p1point > p2point AndAlso p2point >= 3 Then
            score = "Advantage player1"
        End If

        If p2point > p1point AndAlso p1point >= 3 Then
            score = "Advantage player2"
        End If

        If p1point >= 4 AndAlso p2point >= 0 AndAlso (p1point - p2point) >= 2 Then
            score = "Win for player1"
        End If

        If p2point >= 4 AndAlso p1point >= 0 AndAlso (p2point - p1point) >= 2 Then
            score = "Win for player2"
        End If

        Return score
    End Function

    Public Sub SetP1Score(ByVal number As Integer)
        For i As Integer = 0 To number - 1
            P1Score()
        Next
    End Sub

    Public Sub SetP2Score(ByVal number As Integer)
        For i = 0 To number - 1
            P2Score()
        Next
    End Sub

    Private Sub P1Score()
        p1point += 1
    End Sub

    Private Sub P2Score()
        p2point += 1
    End Sub

    Public Sub WonPoint(ByVal player As String) Implements ITennisGame.WonPoint
        If player = "player1" Then
            P1Score()
        Else
            P2Score()
        End If
    End Sub
End Class
